webapp/channels/src/components/drafts/thread_draft/thread_draft.tsx

```diff
@@ -58,18 +58,18 @@ function ThreadDraft({
 
     const onSubmit = useMemo(() => {
         if (thread) {
-            return makeOnSubmit(channel.id, thread.id, '');
+            return makeOnSubmit(channel!.id, thread.id, '');
         }
 
         return () => Promise.resolve({data: true});
-    }, [channel.id, thread?.id]);
+    }, [channel?.id, thread?.id]);
 
     const handleOnDelete = useCallback((id: string) => {
-        dispatch(removeDraft(id, channel.id, rootId));
-    }, [channel.id, rootId]);
+        dispatch(removeDraft(id, channel!.id, rootId));
+    }, [channel?.id, rootId]);
 
     const handleOnEdit = useCallback(() => {
-        dispatch(selectPost({id: rootId, channel_id: channel.id} as Post));
+        dispatch(selectPost({id: rootId, channel_id: channel!.id} as Post));
```

Comment:
Should we be asserting it here? What happens if it's `undefined`?

Reply:
Short answer: It will crash.

Long answer:
My train of thought here is:
- handleOnEdit is only called by the DraftActions
- DraftActions is only rendered if channel is defined
- channel will always be defined when this is called

This is not transparent for the one reading the code, and that makes it a bad decision, but I was a bit tired of so much boilerplate :sweat_smile:

Reply:
See this comment: https://github.com/mattermost/mattermost/pull/26393#discussion_r1521488041
We'll have to continue the discussion on how to properly handle null/undefined I think.

Reply:
Can we get these channel IDs from the draft prop? We need the rest of the channel object for rendering, but these actions all use the ID